### ISSUE TITLE
Great article thanks ... should setSecret and testSecret throw errors?

### ISSUE THREAD

**Issue by vespertilian** [matt-tyler/secret-rotation-example#1]
This did not seem to work for me ... as the rotation would throw the error because these two functions still get called right?
Should have this been a console.log() ? 

```
async function setSecret(client, arn, token) {
    throw new Error("Not Implemented");
}

async function testSecret(client, arn, token) {
    throw new Error("Not Implemented")
}
```

**Comment by matt-tyler** [comment]
If I’m being entirely honest, I can’t remember! I know it’s not necessary
to implement those functions in certain situations- but I can’t remember if
those functions got executed as part of side effects in other steps or
whether they are always executed.

I did actually run this code... so I thought it was the former. I do have a
live version of this running somewhere that was based on the example. I’ll
double check tomorrow and get back to you!

**Comment by vespertilian** [comment]
It was just sitting throwing errors for me... I hope your key has been rotating. :)

**Comment by matt-tyler** [comment]
Yep - I've had a look and you are correct - they not meant to be throwing errors.
